app/lib/cms/sts/assume_role_web_identity.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module CMS
+  module STS
+    class AssumeRoleWebIdentity
+      include Singleton
+
+      class TokenNotFoundError < StandardError
+      end
+
+      def identity_credentials
+        @identity_credentials ||= begin
+          check_token_presence
+
+          Aws::AssumeRoleWebIdentityCredentials.new(
+            region: config[:region],
+            role_arn: config[:role_arn],
+            role_session_name: config[:role_session_name],
+            web_identity_token_file: config[:web_identity_token_file]
```

Comment:
Correct me if I'm wrong but when we are here, we are inside the `CMS::S3.credentials` call. This should still allow us to get the rest of the values though from it. So this suggestion makes sense to me unless I miss something and we go into an infinite loop.

Reply:
For this change, we'd have to expose `role_arn`, `role_session_name` and `web_identity_token_file` on `CMS::S3`. I'm wondering if that'd be the best approach, since none of those values would be usable without each other... Maybe expose a "sts_arguments" with all those values?

With the current implementation, the single source of truth is the `amazon_s3.yml` config file, witch we access through the Application variable.

No real preference from my side... What do you think would be better?

Reply:
I think either are good. I'd prefer exposing the three properties in different methods, but I don't know why 😅. It just looks nicer to me.

Reply:
Updated in 17a7ad1.